WordPress/gutenberg
# Remove block renaming control from advanced inspector controls group

## What?
Closes #60118

As proposed in that PR, removes the block renaming input from the advanced section of inspector controls.

## How?
Highlight the code, press delete key

## Testing Instructions
1. Add a block that supports renaming (paragraph, group)
2. Try renaming it in the advanced section, find out you can't
3. Try renaming it via list view, it should still work

## Screenshots or screencast 
### Before
![Screenshot 2024-04-04 at 2 25 08 pm](https://github.com/WordPress/gutenberg/assets/677833/7e7aea6d-706e-4ab6-a422-e87ef78609a9)

### After
![Screenshot 2024-04-04 at 2 26 03 pm](https://github.com/WordPress/gutenberg/assets/677833/b3537a2d-c4d2-4bdb-a48e-29ff9575414f)
